src/utils/OpenProXLLF.cpp

```diff
@@ -0,0 +1,1007 @@
+      }
+    }
+    for (Size k = 0; k < cross_link_residue2.size(); k++)
+    {
+      if (cross_link_residue2[k] == "K")
+      {
+        n_term_linker = true;
+      }
+      if (cross_link_residue2[k] == "D")
+      {
+        c_term_linker = true;
+      }
+    }
+
+    // lookup for processed peptides. must be defined outside of omp section and synchronized
+    vector<OpenProXLUtils::PeptideMass> peptide_masses;
+
+    Size count_proteins = 0;
+    Size count_peptides = 0;
+
+    progresslogger.startProgress(0, 1, "Digesting peptides...");
+    peptide_masses = OpenProXLUtils::digestDatabase(fasta_db, digestor, min_peptide_length, cross_link_residue1, cross_link_residue2, fixed_modifications,  variable_modifications, max_variable_mods_per_peptide, count_proteins, count_peptides, n_term_linker, c_term_linker);
+    progresslogger.endProgress();
+
+    // create spectrum generator
+    TheoreticalSpectrumGeneratorXLMS specGen;
+
+    // Setting parameters for cross-link fragmentation
+    Param specGenParams = specGen.getParameters();
+    specGenParams.setValue("add_isotopes", "true", "If set to 1 isotope peaks of the product ion peaks are added");
+    specGenParams.setValue("max_isotope", 2, "Defines the maximal isotopic peak which is added, add_isotopes must be set to 1");
+    specGenParams.setValue("add_losses", "false", "Adds common losses to those ion expect to have them, only water and ammonia loss is considered");
+    specGenParams.setValue("add_precursor_peaks", "false", "Adds peaks of the precursor to the spectrum, which happen to occur sometimes");
+    specGenParams.setValue("add_abundant_immonium_ions", "false", "Add most abundant immonium ions");
+    specGenParams.setValue("add_first_prefix_ion", "true", "If set to true e.g. b1 ions are added");
+    specGenParams.setValue("add_y_ions", "true", "Add peaks of y-ions to the spectrum");
+    specGenParams.setValue("add_b_ions", "true", "Add peaks of b-ions to the spectrum");
+    specGenParams.setValue("add_a_ions", "false", "Add peaks of a-ions to the spectrum");
+    specGenParams.setValue("add_c_ions", "false", "Add peaks of c-ions to the spectrum");
+    specGenParams.setValue("add_x_ions", "false", "Add peaks of  x-ions to the spectrum");
+    specGenParams.setValue("add_z_ions", "false", "Add peaks of z-ions to the spectrum");
+    // TODO does nothing yet
+    specGenParams.setValue("multiple_fragmentation_mode" , "false", "If set to true, multiple fragmentation events on the same cross-linked peptide pair are considered (HCD fragmentation)");
+    specGen.setParameters(specGenParams);
+
+    LOG_DEBUG << "Peptide candidates: " << peptide_masses.size() << endl;
+    search_params = protein_ids[0].getSearchParameters();
+    search_params.setMetaValue("MS:1001029", peptide_masses.size()); // number of sequences searched = MS:1001029
+    protein_ids[0].setSearchParameters(search_params);
+
+    vector<OpenProXLUtils::XLPrecursor> enumerated_cross_link_masses;
```

Comment:
it looks like there are several hundred lines of code duplicated between `OpenProXL.cpp` and `OpenProXLLF.cpp`